### PR TITLE
rsa: bugfix about code framework

### PR DESCRIPTION
**Pull request by CaptainMiaow**
This patch including:
1. support switch to software algs
2. optimize prime generation and prime check functions
3. optimize the code of keygen/enc/dec/sign/verify
4. bugfix about unreleased resources
5. unify the name style of functions
6. cleanup code

Signed-off-by: Zhiqi Song <songzhiqi1@huawei.com>

### CODE REVIEW

**Comment by gaozhangfei** [comment]
why change api?
-               uadk_destroy_rsa();
+               uadk_e_destroy_rsa();

File: src/uadk_rsa.c [hunk 1]
```diff
@@ -604,161 +713,200 @@ static int uadk_wd_rsa_init(struct rsa_res_config *config,
...
+{
+	struct uacce_dev *dev;
+	int ret;
+
+	if (g_rsa_res.pid != getpid()) {
+		pthread_spin_lock(&g_rsa_res.lock);
+		if (g_rsa_res.pid == getpid()) {
+			pthread_spin_unlock(&g_rsa_res.lock);
+		return 0;
+	}
```
Comment (gaozhangfei) [inline src/uadk_rsa.c:726]: tab?

**Review by gaozhangfei — COMMENTED** [review]
still see return -1 and return MACRO, is that need aligned?

File: src/uadk_rsa.c (hunk 1 above)
Comment (CaptainMiaow) [reply src/uadk_rsa.c:726]: > why change api?
> 
> * ```
>             uadk_destroy_rsa();
>   ```
> 
> * ```
>             uadk_e_destroy_rsa();
>   ```

Code inspectors think the uadk_ prefix is not clearly expressed, may represent uadk driver, and suggest using the uadk_e prefix to define the API, e is the abbr. of engine.

Comment (CaptainMiaow) [reply src/uadk_rsa.c:726]: > tab?

OK, I will make a change, tks.

**Comment by CaptainMiaow** [comment]
> still see return -1 and return MACRO, is that need aligned?

That's a problem,  I will make a change, tks.

**Comment by CaptainMiaow** [comment]
> why change api?
> 
> * ```
>             uadk_destroy_rsa();
>   ```
> 
> * ```
>             uadk_e_destroy_rsa();
>   ```

Code inspectors think the uadk_ prefix is not clearly expressed, may represent uadk driver, and suggest using the uadk_e prefix to define the API, e is the abbr. of engine.

File: src/uadk_rsa.c (hunk 1 above)
Comment (CaptainMiaow) [reply src/uadk_rsa.c:726]: > tab?
OK, I will make a change, tks.

**Comment by gaozhangfei** [comment]
uadk_rsa.c:950:12: warning: ‘uadk_rsa_sync’ defined but not used [-Wunused-function]

**Comment by CaptainMiaow** [comment]
> uadk_rsa_sync

OK, I fixed it.

**Pull request merged**